Fixed #30947 -- Changed tuples to lists in model Meta options in django.contrib modules.

jdufresne opened the pull request:
The Django "Model Meta options" docs provide examples and generally point the reader to use lists for the `unique_together` and `ordering` options. Follow our own advice for contrib models.

More generally, lists should be used for homogeneous sequences of arbitrary lengths of which both `unique_together` and `ordering` are.

https://code.djangoproject.com/ticket/30947

jdufresne commented:
The choice of list is explained quite well in #11267 and acknowledges these differences.

jdufresne commented:
Added ticket https://code.djangoproject.com/ticket/30947

codingjoe commented:
@jdufresne one last thing, even tho this is a small change, it's best to add at least a one liner to the release notes. `3.1.txt` would be the right one.

felixxm commented:
@jdufresne Thanks :+1: I added a separate commit with small docs changes and added release notes.

Migrations are required for changes in `Meta.ordering`.

jdufresne commented:
Updated the migrations.

As there is no actual schema change (or behavioral change really), I modified the old migrations instead of creating new ones. This avoids unnecessary processing of additional migrations.

felixxm commented on django/contrib/contenttypes/migrations/0001_initial.py:
```diff
@@ -17,7 +17,7 @@ class Migration(migrations.Migration):
             ],
             options={
-                'ordering': ('name',),
+                'ordering': ['name'],
```
This change is unrelated, I'm going to revert it.

jdufresne replied:
No problem, but do you mind explaining a bit? Above you requested changes to the migrations as well.
